scripts/dts/gen_defines.py

```diff
@@ -1097,6 +1106,13 @@ def quote_str(s):
     return f'"{escape(s)}"'
 
 
+def escape_line_breaks(s):
+    # Prepend a line continuation character to any new lines in 's' so all the
+    # lines are kept together in a #define value.
+
+    return s.replace("\r", "\\\r").replace("\n", "\\\n")
```

Comment:
to be clear, if the property is "first\nsecond", then the macro for string_unquoted version will be `firstsecond` with this logic. is this intentional?

Reply:
Looking at this again, I realized that this doesn't actually insert a line break into the expanded macro, because that is not possible to begin with.

Would inserting a space be a reasonable alternative? If I understand this correctly, it is used to expand the string into something that can be evaluated as C code, in which case all that matters is that there _is_ whitespace, not what kind.

Reply:
I'm not really sure what it should be, since I don't know a use case where you would or wouldn't want to know about  a newline in the unquoted version of the string, I just wanted to be clear if you were aware that's what it was doing. I just don't think we should leave the behavior undefined

Reply:
Given the limitations of C macros, I don't believe it's possible to know about a new line in the unquoted string. I've changed it to be replaced by a space and added some tests for using `\n` and `\"` in unquoted strings. 

I cannot think of any reason you would intentionally use `\n` with the unquoted string API, but it at the very least needs to generate code that compiles so it can be used in quoted strings, so I just duplicated one of the existing tests but with `\n` in place of spaces.

`\"` is plausible to intentionally use with the unquoted API, since it could be used for joining multiple string literals, so I've included a test for that as well.